crates/uv/src/requirements.rs

```diff
@@ -512,43 +511,46 @@ pub(crate) struct NamedRequirements {
     pub(crate) overrides: Vec<Requirement>,
     /// Package to install as editable installs
     pub(crate) editables: Vec<EditableRequirement>,
-    /// The index URL to use for fetching packages.
-    pub(crate) index_url: Option<IndexUrl>,
-    /// The extra index URLs to use for fetching packages.
-    pub(crate) extra_index_urls: Vec<IndexUrl>,
-    /// Whether to disallow index usage.
-    pub(crate) no_index: bool,
-    /// The `--find-links` locations to use for fetching packages.
-    pub(crate) find_links: Vec<FlatIndexLocation>,
 }
 
 impl NamedRequirements {
     /// Convert a [`RequirementsSpecification`] into a [`NamedRequirements`].
-    pub(crate) fn from_spec(spec: RequirementsSpecification) -> Result<Self> {
-        Ok(Self {
-            project: spec.project,
-            requirements: spec
-                .requirements
-                .into_iter()
-                .map(|requirement| match requirement {
+    pub(crate) async fn from_spec(
+        requirements: Vec<RequirementsTxtRequirement>,
+        constraints: Vec<Requirement>,
+        overrides: Vec<Requirement>,
+        editables: Vec<EditableRequirement>,
+        cache: &Cache,
+        client: &RegistryClient,
+    ) -> Result<Self> {
+        // Resolve all unnamed references.
+        let requirements = futures::stream::iter(requirements)
+            .map(|requirement| async {
+                match requirement {
                     RequirementsTxtRequirement::Pep508(requirement) => Ok(requirement),
                     RequirementsTxtRequirement::Unnamed(requirement) => {
-                        Self::name_requirement(requirement)
+                        Self::name_requirement(requirement, cache, client).await
                     }
-                })
-                .collect::<Result<_>>()?,
-            constraints: spec.constraints,
-            overrides: spec.overrides,
-            editables: spec.editables,
-            index_url: spec.index_url,
-            extra_index_urls: spec.extra_index_urls,
-            no_index: spec.no_index,
-            find_links: spec.find_links,
+                }
+            })
+            .buffer_unordered(50)
```

Comment:
Do we need to sort those? iirc the order here determines the prioritization in pubgrub.

Reply:
Good call...